nixos/modules/services/network-filesystems/iroh.nix

```diff
@@ -0,0 +1,275 @@
+{ config, lib, pkgs, utils, ... }:
+with lib;
+let
+  # NB: build the manual with:
+  # $ nix-build nixos/release.nix -A manual.x86_64-linux
+
+  # NB: run the tests with:
+  # $ nix-build -A nixosTests.iroh
+
+  # Docs for writing tests:
+  # https://nixos.org/manual/nixos/stable/index.html#sec-nixos-tests
+
+  # TODO test: configure systemd logs directory:
+  # iroh: Failed to create log directory: Read-only filesystem
+
+  # TODO run systemd-analyze security and apply whatever sandboxing doesn't break the service
+  # TODO add tests
+
+  # cfg is a conventional alias for the configuration of the current module
+  cfg = config.services.iroh;
+
+  # Default listening port.
+  listenPort = 11204;
+  # Default RPC port.
+  rpcPort = 1337;
+  # Default metrics port. With a value of -1 serving metrics is disabled.
+  metricsPort = -1;
```

Comment:
no need to set these values here, you can just set them in `default` values below and reference them elsewhere as needed